ui/components/multichain/network-list-menu/network-list-menu.tsx

```diff
@@ -291,78 +316,147 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
+    } else {
+      await handleNonEvmNetworkChange(chainId);
+    }
+
+    const chainIdToTrack = isEvm ? convertCaipToHexChainId(chainId) : chainId;
+    const currentChainIdToTrack = isEvm
+      ? convertCaipToHexChainId(currentChainId)
+      : currentChainId;
+
     trackEvent({
       event: MetaMetricsEventName.NavNetworkSwitched,
       category: MetaMetricsEventCategory.Network,
       properties: {
         location: 'Network Menu',
-        chain_id: currentChainId,
-        from_network: currentChainId,
-        to_network: network.chainId,
+        chain_id: currentChainIdToTrack,
+        from_network: currentChainIdToTrack,
+        to_network: chainIdToTrack,
       },
     });
   };
 
+  const getNetworkFlags = (
+    network: MultichainNetworkConfiguration,
+  ): Record<string, boolean> => {
+    if (!network.isEvm) {
+      return {
+        isEnabled: isAccountInNetwork(network.chainId) || isUnlocked,
+        isDeletable: false,
+        isEditable: false,
+        hasMultiRpcOptions: false,
+      };
+    }
+
+    const rpcEndpoints =
+      getRpcDataByChainId(network.chainId, evmNetworks)?.rpcEndpoints || [];
+    return {
+      isEnabled: true,
+      isDeletable:
+        isUnlocked &&
+        network.chainId !== currentChainId &&
+        network.chainId !== EthScope.Mainnet,
+      isEditable: true,
+      hasMultiRpcOptions: rpcEndpoints.length > 1,
+    };
+  };
+
   // Renders a network in the network list
-  const generateNetworkListItem = (network: NetworkConfiguration) => {
+  const generateMultichainNetworkListItem = (
+    network: MultichainNetworkConfiguration,
+  ) => {
     const isCurrentNetwork = network.chainId === currentChainId;
-    const canDeleteNetwork =
-      isUnlocked && !isCurrentNetwork && network.chainId !== CHAIN_IDS.MAINNET;
+    const { isEnabled, isDeletable, isEditable, hasMultiRpcOptions } =
+      getNetworkFlags(network);
+
+    const onDelete = () => {
+      dispatch(toggleNetworkMenu());
+      dispatch(
+        showModal({
+          name: 'CONFIRM_DELETE_NETWORK',
+          // Non-EVM networks cannot be deleted, so it's safe to call
+          // this conversion function here.
+          target: convertCaipToHexChainId(network.chainId),
+          onConfirm: () => undefined,
+        }),
+      );
+    };
+
+    const onEdit = () => {
+      dispatch(
+        setEditedNetwork({
+          // Non-EVM networks cannot be edited, so it's safe to call
+          // this conversion function here.
+          chainId: convertCaipToHexChainId(network.chainId),
+          nickname: network.name,
+        }),
+      );
+      setActionMode(ACTION_MODE.ADD_EDIT);
+    };
+
+    const onRpcConfigEdit = () => {
+      setActionMode(ACTION_MODE.SELECT_RPC);
+      dispatch(
+        // Non-EVM networks cannot be edited, so it's safe to call
+        // this conversion function here.
+        setEditedNetwork({ chainId: convertCaipToHexChainId(network.chainId) }),
+      );
+    };
```

Comment:
We could make this more DRY maybe: (we could use `hexChainId` to match the names of other functions if you prefer)

```suggestion
    // Non-EVM networks cannot be edited, so it's safe to call
    // this conversion function here.
    const chainId = convertCaipToHexChainId(network.chainId);

    const onDelete = () => {
      dispatch(toggleNetworkMenu());
      dispatch(
        showModal({
          name: 'CONFIRM_DELETE_NETWORK',
          target: chainId,
          onConfirm: () => undefined,
        }),
      );
    };

    const onEdit = () => {
      dispatch(
        setEditedNetwork({
          chainId,
          nickname: network.name,
        }),
      );
      setActionMode(ACTION_MODE.ADD_EDIT);
    };

    const onRpcConfigEdit = () => {
      setActionMode(ACTION_MODE.SELECT_RPC);
      dispatch(setEditedNetwork({ chainId }));
    };
```

Reply:
my idea https://github.com/MetaMask/metamask-extension/pull/30433/commits/3bb12c03976e2337f9b71609774d5a52fb039926

Reply:
there was a bug. Fixed here [f7f95b0](https://github.com/MetaMask/metamask-extension/pull/30433/commits/f7f95b05998cfa54962803c0c510873ce422c2d6)

Reply:
Looks perfect to me 👍 thanks! 😄